src/lib.rs

```diff
@@ -1037,15 +1037,21 @@ pub mod pallet {
 		///
 		/// - `query_id`: Query identifier being disputed.
 		/// - `timestamp`: Timestamp being disputed.
+		/// - 'beneficiary`: beneficiary address of user to receive the slash winning
 		#[pallet::call_index(7)]
 		pub fn begin_dispute(
 			origin: OriginFor<T>,
 			query_id: QueryId,
 			timestamp: Timestamp,
+			beneficiary: Option<Address>,
 		) -> DispatchResult {
 			let dispute_initiator = ensure_signed(origin)?;
-			// Only reporters can begin disputes due to requiring an account on staking chain to potentially receive slash amount if dispute successful
-			ensure!(<StakerDetails<T>>::contains_key(&dispute_initiator), Error::<T>::NotReporter);
+
+			// the Reporter can only be validated when address field is None
```

Comment:
Change comment to something like 'use dispute initiator's remote address (if available) when no beneficiary address specified' please.